host_core/test/support/common.exs

```diff
@@ -137,4 +137,46 @@ defmodule HostCoreTest.Common do
       policy_timeout_ms: 1_000
     }
   end
+
+  # Benchmarking common functions
+  # Helper function to run before benchmark tests
+  def pre_benchmark_run() do
+    # Set level to info to reduce log noise
+    Logger.configure(level: :info)
+  end
+
+  # Helper function to run after benchmark tests
+  def post_benchmark_run() do
+    # Return log level to debug
+    Logger.configure(level: :debug)
+  end
+
+  @spec run_benchmark(
+          test_config :: map(),
+          num_actors :: non_neg_integer(),
+          parallel :: list() | non_neg_integer(),
+          warmup :: non_neg_integer(),
+          time :: non_neg_integer()
+        ) :: :ok
+  # Run a benchmark with specified config, repeating for each parallel argument if it's a list
+  def run_benchmark(test_config, num_actors, parallel \\ [1], warmup \\ 1, time \\ 5)
+      when is_list(parallel) do
+    parallel
+    |> Enum.each(fn p -> run_benchmark(test_config, num_actors, p, warmup, time) end)
+
+    :ok
+  end
+
+  def run_benchmark(test_config, num_actors, parallel, warmup, time)
```

Comment:
Should `run_benchmark` call the pre/post functions so the caller doesn't have to explicitly adjust log levels? Or do you think there are cases we wouldn't want to set the level to info during a benchmark?

Reply:
Actually, pretty fair. I'll make it call the functions and we can split that out if it's needed later 😄